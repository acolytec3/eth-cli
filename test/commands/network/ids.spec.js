const NetworkIdsCommand = require('../../../src/commands/network/ids')

describe('network:ids', () => {
  let stdoutResult

  beforeEach(() => {
    stdoutResult = []
    jest
      .spyOn(process.stdout, 'write')
      .mockImplementation(val => stdoutResult.push(require('strip-ansi')(val.toString('utf8'))))
  })

  afterEach(() => jest.restoreAllMocks())

  it(`Should run 'network:ids --help' and throw an error.`, async () => {
    await expect(NetworkIdsCommand.run(['--help'])).rejects.toThrow('EEXIT: 0')
  })

  it(`Should run 'network:ids' and match snapshot.`, async () => {
    await NetworkIdsCommand.run([])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'network:ids --display=json' and match snapshot.`, async () => {
    await NetworkIdsCommand.run(['--display=json'])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'network:ids --display=table' and match snapshot.`, async () => {
    await NetworkIdsCommand.run(['--display=table'])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'network:ids --bar' and throw an error.`, async () => {
    await expect(NetworkIdsCommand.run(['--bar'])).rejects.toThrow('Unexpected argument: --bar')
  })
})
