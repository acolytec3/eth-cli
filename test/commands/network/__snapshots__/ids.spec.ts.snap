// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`network:ids Should run 'network:ids --display=json' and match snapshot. 1`] = `
Array [
  "{
  \\"mainnet\\": 1,
  \\"ropsten\\": 3,
  \\"rinkeby\\": 4,
  \\"rsk\\": 30,
  \\"rsk-testnet\\": 31,
  \\"kovan\\": 42,
  \\"sokol\\": 77,
  \\"poa\\": 99,
  \\"xdai\\": 100
}
",
]
`;

exports[`network:ids Should run 'network:ids --display=table' and match snapshot. 1`] = `
Array [
  "┌─────────┬─────────┬─────────┬─────┬─────────────┬───────┬───────┬─────┬──────┐
│ mainnet │ ropsten │ rinkeby │ rsk │ rsk-testnet │ kovan │ sokol │ poa │ xdai │
├─────────┼─────────┼─────────┼─────┼─────────────┼───────┼───────┼─────┼──────┤
│ 1       │ 3       │ 4       │ 30  │ 31          │ 42    │ 77    │ 99  │ 100  │
└─────────┴─────────┴─────────┴─────┴─────────────┴───────┴───────┴─────┴──────┘
",
]
`;

exports[`network:ids Should run 'network:ids' and match snapshot. 1`] = `
Array [
  "{
  \\"mainnet\\": 1,
  \\"ropsten\\": 3,
  \\"rinkeby\\": 4,
  \\"rsk\\": 30,
  \\"rsk-testnet\\": 31,
  \\"kovan\\": 42,
  \\"sokol\\": 77,
  \\"poa\\": 99,
  \\"xdai\\": 100
}
",
]
`;
